app/js/components/common/header.js

```diff
@@ -0,0 +1,146 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+
+import React, {Component} from 'react';
+import {Link, IndexLink} from 'react-router';
+import {ApiHelper} from '../../helpers/apiHelper';
+
+const NUMBER_OF_COLUMNS = 3;
+
+export default class Header extends Component {
+  constructor() {
+    super();
+    this.state = {
+      locationTags: [],
+      currentLocationTag: "",
+      defaultLocation: "",
+      currentUser: "",
+      currentLogOutUrl: "",
+    };
+    this.getUri = this.getUri.bind(this);
+  }
+
+  componentWillMount() {
+    this.fetchLocation('/location').then((response) => {
+      this.setState({locationTags: response.results});
+      this.setState({defaultLocation: response.results[0].display});
+      this.getUri();
+    });
+
+    this.fetchLocation('/session').then((response) => {
+      this.setState({currentUser: response.user.display});
+    });
+  }
+
+  getLocations() {
+    return this.state.locationTags.map((location) => {
+      return location.display;
+    });
+  }
+
+  getUri() {
+    this.state.locationTags.map((location) => {
+      let url = location.links[0].uri;
+      let arrUrl = url.split("/");
+      let applicationInUse = arrUrl[3].search('http:') == -1 ? arrUrl[3]: arrUrl[3].replace('http:', '');
+      let customUrl = `/${applicationInUse}/appui/header/logout.action?successUrl=${applicationInUse}`;
+      this.setState({currentLogOutUrl: customUrl});
+      return customUrl;
+        
+    });
+  }
+
+  fetchLocation(url) {
+    const apiHelper = new ApiHelper(null);
+    const getData = new Promise(function(resolve, reject) {
+      apiHelper.get(url).then(response => {
```

Comment:
Are other groups aware of these utility classes like apiHelper?

Reply:
Yes, they are. The August group is using (this)[https://github.com/openmrs/openmrs-owa-obsadmin/blob/master/app/js/utilities/apiHelper.js].

Reply:
Awesome. Please share as many goodies as you can. 😊